rest/model/admin.go

```diff
@@ -0,0 +1,118 @@
+package model
+
+import (
+	"errors"
+
+	"github.com/evergreen-ci/evergreen/model/admin"
+)
+
+// APIAdminSettings is the structure of a response to the admin route
+type APIAdminSettings struct {
+	Banner       APIString       `json:"banner"`
+	ServiceFlags APIServiceFlags `json:"service_flags"`
+}
+
+// APIBanner is a public structure representing the banner part of the admin settings
+type APIBanner struct {
+	Text APIString `json:"banner"`
+}
+
+// APIServiceFlags is a public structure representing the admin service flags
+type APIServiceFlags struct {
+	TaskDispatchDisabled  bool `json:"task_dispatch_disabled"`
+	HostinitDisabled      bool `json:"hostinit_disabled"`
+	MonitorDisabled       bool `json:"monitor_disabled"`
+	NotificationsDisabled bool `json:"notifications_disabled"`
+	AlertsDisabled        bool `json:"alerts_disabled"`
+	TaskrunnerDisabled    bool `json:"taskrunner_disabled"`
+	RepotrackerDisabled   bool `json:"repotracker_disabled"`
+	SchedulerDisabled     bool `json:"scheduler_disabled"`
+}
+
+// BuildFromService builds a model from the service layer
+func (as *APIAdminSettings) BuildFromService(h interface{}) error {
```

Comment:
is there a way that you can have the APIAdminSettings and APIServiceFlags converters share code so you can only write each conversion only once to avoid needing to add a single new type in lots of different contexts.

Reply:
Done